### ISSUE TITLE
Simple Sites: wp-admin/site-editor indefinitely loads default homepage on private sites with custom mapped domains

### ISSUE THREAD

**Issue by dsas** [Automattic/wp-calypso#74958]
### Quick summary

The wp-admin version of the site-editor doesn't load a template when visited. The calypso site editor (wordpress.com/site-editor<siteslug>) does load a template though.

### Steps to reproduce
_updated June 27, 2023_

_The requirements are:_

1. _A block theme that enables the Site Editor_
2. _Default homepage_
3. _Private site (not Coming Soon, but Private)_
4. _A mapped domain_

➡️ Go to `sitename.wordpress.com/wp-admin/site-editor.php`

Open Appearance > Editor 

### What you expected to happen

The site-editor to open and show the template used on my homepage

### What actually happened

The site-editor opened but didn't load a template, the spinner is permanently displayed instead.

### Impact

Some (< 50%)

### Available workarounds?
_updated June 27, 2023_

Yes, easy to implement:

1. _Manually load the desired Home template (Templates --> Home)_
2. _Temporarily set the *.wordpress.com domain to primary_
3. _Temporarily set the site to public_

### Platform (Simple and/or Atomic)

Simple

### Logs or notes

Used Firefox but it's probably not a browser specific issue.

There are no errors in the browser console.

Also reported here: pdtkmj-1jE-p2#comment-2437 and pdtkmj-1jE-p2#comment-2438

I had a quick test and can't reproduce it on atomic.

**My Atomic site redirects from /wp-admin/site-editor.php to /wp-admin/site-editor.php?postType=page&postId=2 my simple sites do not. The URL doesn't change unless I explicitly choose a template**

https://user-images.githubusercontent.com/93301/227960728-d38c4b1a-4e93-4bdc-90fc-106ef0e82f45.mp4

**Comment by obenland** [comment]
@mattwiebe Flagging for your since you're currently working on enabling the Site Editor for all Simple sites.

**Comment by dsas** [comment]
This doesn't appear to be every simple site, it might depend on the theme. I could reproduce it on TT2, TT1 and a bevy of other themes but couldn't reproduce it on Geologist and Quadrat

Reproduced on:
 * deantest932.wordpress.com (Blank Canvas 3)
 * deanpremiumtest.wordpress.com (Blank Canvas 3)
 * deantest70.wordpress.com (Vetro)
 * deantest84.wordpress.com (Bibimbap)
 * deantest029.wordpress.com (Zoologist switched to Geologist Blue)


But couldn't reproduce on:
 * deansas.org (Geologist)
 * deantest49.wordpress.com (Quadrat)

**Comment by mattwiebe** [comment]
> @mattwiebe Flagging for your since you're currently working on enabling the Site Editor for all Simple sites.

@obenland - apologies for missing this - I wasn't working on that exactly but it might be that we need to add some query args in our outbound/iframe URLs

**Comment by essleeung** [comment]
I ran into this on this user's simple site- 6070040-zd-woothemes

**Comment by davewhitley** [comment]
I believe I ran into this too. I can reproduce by going through onboarding, picking a block theme, and then edit site design via Launchpad. The page I land on (/wp-admin/site-editor.php?canvas=edit) gets stuck loading.

**Comment by cuemarie** [comment]
📌 SCRUBBING : RESULT - *Replicated / Could Not Replicate / Uncertain*
- Tested on Simple – **Replicated**

📌 FINDINGS/SCREENSHOTS/VIDEO
- I'm able to replicate the problem when the homepage is set to `--Default--` on an FSE theme. In that case, `SITE_NAME/wp-admin/site-editor.php` fails to load.  Replicated in the following themes:
     - Twenty Twenty-Three
     - Zoologist
     - Twenty Twenty-Two
     - Blank Canvas 

On the same sites using those themes, if a static page is set as the homepage, then both Calypso and WP-Admin Site editors open without issue.

**Screenshots/Recordings**

![Markup on 2023-04-05 at 13:11:31](https://user-images.githubusercontent.com/27249804/230198723-3e327590-1f42-406f-89db-ba0ae6be31b1.png)

📌 ACTIONS
- Assigned to Cylon

**Comment by ClassicRKR27** [comment]
I have another report of this here: 42447811-hc

**Comment by happychait** [comment]
42685362-hc

**Comment by happychait** [comment]
39163001-hc

**Comment by dragstor** [comment]
6219933-zen

Simple site + Zoologist

Shared this workaround with the user: 

https://user-images.githubusercontent.com/363811/233783723-61e9815c-ca32-4964-83e4-b6d99b305896.mp4

**Comment by kspilarski** [comment]
The same case for 6223270-zd.

**Comment by kspilarski** [comment]
Another report in 42821028-hc.

**Comment by dsas** [comment]
https://github.com/Automattic/wp-calypso/issues/76105 might be related to an uptick of occurrences of this issue, though it's slightly different - I could reproduce this on *.wordpress.com domains and before the editor was uniframed.

I can't reproduce this issue now 🤔

**Comment by dragstor** [comment]
Reported in 6231240-zen but I'm unable to reproduce it.

**Comment by jartes** [comment]
Another one on 6314836-zd-woothemes that I was able to reproduce with SSP. 

Simple site with the `Stewart` theme.

Edit: it seems that when site is set to Coming Soon / Public, it works. However, when is set to Private, they get this issue.

Shared the workaround from @dragstor, thanks for that!

**Comment by edwinho89** [comment]
6322944-zd-woothemes

Simple site on `Loudness` theme.

Workaround by managing all templates from `/wp-admin/site-editor.php?path=%2Fwp_template%2Fall` then select the homepage template.

**Comment by rw-ye** [comment]
6322777-zd-woothemes

Simple site
Pendant theme

**Comment by pauloeaquino** [comment]
6322842-zd

**Comment by christianguerrag** [comment]
Reported here 6346236-zen but they decided to use a different theme because of this.

**Comment by christianguerrag** [comment]
Reported here 6402842-zen

**Comment by cuemarie** [comment]
@mattwiebe with the more recent updates to Cylon, is this still something that's on your radar, or should we see if someone else can have a look? Still seem to be getting periodic reports from customers about it. Thanks!

**Comment by mattwiebe** [comment]
@cuemarie you can keep pinging me for now if more starts coming up, looks pretty quiet right now?

**Comment by cuemarie** [comment]
@mattwiebe sure thing!  I am able to replicate consistently on a Simple Site + TT3 + Private privacy status, which may explain why we aren't seeing a deluge of reports.


https://github.com/Automattic/wp-calypso/assets/27249804/0a74b60b-8acc-4fdf-991b-886fa0a9ffeb

---

📌 Message to Future Reporters
- HEs, can you please note the privacy status of any sites reported in the future?  Curious if this issue is now only impacting Private sites. Thanks!

**Comment by mattwiebe** [comment]
@cuemarie not able to duplicate on a fresh Simple site with TT3 + Private site. Did you have it mapped to a domain at the time? I tried to look at the one you have in the video but it was mapped to the wpcom subdomain and had a non-Site Editor theme enabled (Twenty Twenty) when I looked. If this is a common test site for you then sometimes older sites can get into weird conditions.

Oh and to be clear: the new site I set up is in Coming Soon mode. Is there a different private mode you were using?

Pasting the console and especially any errors is also very useful when trying to debug something like this.

**Comment by cuemarie** [comment]
Hey @mattwiebe ! Thanks for the reply and questions - hope these additional tests help!

>  Did you have it mapped to a domain at the time?

Yes, I had the site mapped to a domain at the time, and I've tested that again on a new site, and that seems to be part of the issue.  More details below.

> I tried to look at the one you have in the video but it was mapped to the wpcom subdomain and had a non-Site Editor theme enabled (Twenty Twenty) when I looked. If this is a common test site for you then sometimes older sites can get into weird conditions.

Yep, that's totally a test site I use a lot, so I've retested on a brand new simple site.

> Oh and to be clear: the new site I set up is in Coming Soon mode. Is there a different private mode you were using?

Yes, Coming Soon mode is different than the Private mode I'm referring to.  After you launch the site, from the same  Settings > General page you can set the site to **Private** - this is the privacy status that seems to be at issue.

> Pasting the console and especially any errors is also very useful when trying to debug something like this.

Happy to share that as well! I'm not quite sure what's best to share, but I've included the Console errors and warnings in my recording below.

---

So for this test, I took the following steps to replicate the problem:

- Create a new site on the Personal plan
- Set up a custom domain (registered thru WPcom and transferred over from another site) as the primary address
- Under Settings > General (`https://wordpress.com/settings/general/SITE_ADDRESS`), launch the site then set the privacy to **Private** 
- Open Settings > Reading and make sure the **Homepage** is set to **Default**

Following those steps, the Site Editor fails to open:

https://github.com/Automattic/wp-calypso/assets/27249804/8166a967-9ef5-4a4a-944a-2bf5e40b7cf6

---

This is similar to a previous issue: https://github.com/Automattic/wp-calypso/issues/76105

But in that case the Privacy status did not have an impact, and in this case, I see no errors in the console related to `CORS`.

**Comment by cuemarie** [comment]
<details>
  <summary>Copy/paste from the console errors and warnings in the above test</summary>
  
```
This page uses the non standard property “zoom”. Consider using calc() in the relevant property values, or using “transform” along with “transform-origin: 0 0”. [site-editor.php](https://SITE_ADDRESS/wp-admin/site-editor.php)
JQMIGRATE: Migrate is installed, version 3.4.0 [_static:366:970](https://s0.wp.com/_static/??-eJyNzkEKwkAMBdALOY7daLsQz1LbtKRkkjHJWHp7R9BlUQh8wn+BxDWHQdiBPaYSMpUZ2aKVuw2K2VHqNgmRrMfFDrFy5IHKCBaXOo8Cun3imJB/opBw1t5hDxdHsr1yRPP4BB5F31UW2iYkqgbU/zlSmKHa3kWDFnZM30du6dqcL23Xnrq2WV7sVWf3)
Some cookies are misusing the recommended “SameSite“ attribute 9
Declaring non-string block descriptions is deprecated since version 6.2. [index.min.js:1:582](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/deprecated/index.min.js?m=1686787612i&ver=5f56b9106e825b0b4ab3)
 ⚠️ A8C Analytics is disabled. [site-editor.php:309:13](https://SITE_ADDRESS/wp-admin/site-editor.php)
Most Tracks and Analytics code won't fire. This is likely because you're proxied. To turn analytics on, please add `?flags=a8c-analytics.on` to the page URL and reload the page. [site-editor.php:310:13](https://SITE_ADDRESS/wp-admin/site-editor.php)
downloadable font: no supported format found (font-family: "Noticons" style:normal weight:400 stretch:100 src index:1) source: (end of source list)
Store "core/interface" is already registered. [index.min.js:1:17678](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/data/index.min.js?m=1686787612i&ver=16f144585d33a494a980)
XHRGET
https://public-api.wordpress.com/wp/v2/users/me?context=edit&_locale=user
[HTTP/2 404 Not Found 134ms]

`wp.data.select( 'core/editor' ).getBlocks` is deprecated since version 5.3 and will be removed in version 6.2. Please use `wp.data.select( 'core/block-editor' ).getBlocks` instead. [index.min.js:1:582](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/deprecated/index.min.js?m=1686787612i&ver=5f56b9106e825b0b4ab3)
Store "crowdsignal-forms/editor" is already registered. [index.min.js:1:17678](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/data/index.min.js?m=1686787612i&ver=16f144585d33a494a980)
Block "crowdsignal-forms/poll" is already registered. [index.min.js:1:92308](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/blocks/index.min.js?m=1686787612i&ver=42530a18270fd7554276)
Block "crowdsignal-forms/vote" is already registered. [index.min.js:1:92308](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/blocks/index.min.js?m=1686787612i&ver=42530a18270fd7554276)
Block "crowdsignal-forms/vote-item" is already registered. [index.min.js:1:92308](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/blocks/index.min.js?m=1686787612i&ver=42530a18270fd7554276)
Block "crowdsignal-forms/applause" is already registered. [index.min.js:1:92308](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/blocks/index.min.js?m=1686787612i&ver=42530a18270fd7554276)
Block "crowdsignal-forms/nps" is already registered. [index.min.js:1:92308](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/blocks/index.min.js?m=1686787612i&ver=42530a18270fd7554276)
Block "crowdsignal-forms/feedback" is already registered. [index.min.js:1:92308](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/blocks/index.min.js?m=1686787612i&ver=42530a18270fd7554276)
Block "crowdsignal-forms/cs-embed" is already registered. [index.min.js:1:92308](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/blocks/index.min.js?m=1686787612i&ver=42530a18270fd7554276)
Block "jetpack/layout-grid" is already registered. [index.min.js:1:92308](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/blocks/index.min.js?m=1686787612i&ver=42530a18270fd7554276)
Block "jetpack/layout-grid-column" is already registered. [index.min.js:1:92308](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/blocks/index.min.js?m=1686787612i&ver=42530a18270fd7554276)
Block "jetpack/rating-priciness" is already registered. [index.min.js:1:92308](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/blocks/index.min.js?m=1686787612i&ver=42530a18270fd7554276)
Block "jetpack/rating-spiciness" is already registered. [index.min.js:1:92308](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/blocks/index.min.js?m=1686787612i&ver=42530a18270fd7554276)
Cookie “tk_tc” has been rejected for invalid domain. [w.js:1:1555](https://stats.wp.com/w.js?ver=202325)
wp.nux is deprecated since version 5.4 and will be removed in version 6.2. Note: wp.components.Guide can be used to show a user guide. [index.min.js:1:582](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/deprecated/index.min.js?m=1686787612i&ver=5f56b9106e825b0b4ab3)
Source map error: Error: request failed with status 404
Resource URL: https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/deprecated/index.min.js?m=1686787612i&ver=5f56b9106e825b0b4ab3
Source Map URL: index.min.js.map

Store "automattic/site" is already registered. [index.min.js:1:17678](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/data/index.min.js?m=1686787612i&ver=16f144585d33a494a980)
Source map error: Error: request failed with status 404
Resource URL: https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/data/index.min.js?m=1686787612i&ver=16f144585d33a494a980
Source Map URL: index.min.js.map

Could not determine slot to register the menu item. [_static:1681:18](https://s0.wp.com/_static/??-eJylkttuwjAMhl9oqYUEjJtpj4JSxyvpclritOXtZ7qCUKVpB66i3/Hn/LENY1IYA1NgSK52NhTwZKxWNCUXM2XoL5EpNX15gp+zC+XBIhXg0TLP+JrUxtugrvxgDcWUqRTQpRCX5X6uu0Z9vXE9cdL4vmjwMQY42oDQVusMdNo5ymclFdmGrjRS87svdFVkS7lTY8Lo13qNSUP6jyrFl6OZs1TKcTqrTBIrfGXEkKtG2iGQSJ3sA0Yu+Bsxnn7LYybNJJ40skIn/X3kdfItmX9bP+Afh7nEjgMFEzPoytFrGSbesu9W52vorYsSlsVhAWax7BfYYGgSA6/+ZbM/PG93m/1u238C5nIiug==)
select( 'core/edit-site' ).isFeatureActive is deprecated since version 6.0. Please use select( 'core/preferences' ).get instead. [index.min.js:1:582](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/deprecated/index.min.js?m=1686787612i&ver=5f56b9106e825b0b4ab3)
select( 'core/edit-site' ).toggleFeature( featureName ) is deprecated since version 6.0. Please use select( 'core/preferences').toggle( 'core/edit-site', featureName ) instead. [index.min.js:1:582](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/deprecated/index.min.js?m=1686787612i&ver=5f56b9106e825b0b4ab3)
Partitioned cookie or storage access was provided to “https://widgets.wp.com/3rd-party-cookie-check/index.html” because it is loaded in the third-party context and dynamic state partitioning is enabled.

Source map error: Error: request failed with status 404
Resource URL: https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/blocks/index.min.js?m=1686787612i&ver=42530a18270fd7554276
Source Map URL: index.min.js.map

Cookie “wpcom-thirdparty-cookie-check” does not have a proper “SameSite” attribute value. Soon, cookies without the “SameSite” attribute or with an invalid value will be treated as “Lax”. This means that the cookie will no longer be sent in third-party contexts. If your application depends on this cookie being available in such contexts, please add the “SameSite=None“ attribute to it. To know more about the “SameSite“ attribute, read https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite [complete.html:16:12](https://widgets.wp.com/3rd-party-cookie-check/complete.html)
Partitioned cookie or storage access was provided to “https://widgets.wp.com/3rd-party-cookie-check/complete.html” because it is loaded in the third-party context and dynamic state partitioning is enabled.

3PC allowed [_static:784:13](https://s0.wp.com/_static/??-eJyNUUFuAyEM/FC9KFWS5lL1KRVgZ8MGMAKzaX9fN9lWUQ6rHIxsPDMey+ZSwHMWymJShxL7GHIzE0mx/rzUJjFn8xmyNy6yPzdDGIQrOBI7TO3FPK+SGHukZuaAxKVSU9h9BRYRNOnJukjQS2SL0HopXOVx1t+g9p3Ffp3CeIoaQtVgaLKYHVwPEVdsXgKOJLrUrA1Xg9Cvpduv2kkhr5AzCy2vtlPiDFEVhmdYV21wtsL8ugK2Bw/CHBV4nz8ybmr/SxVVOBKh0wso9CO9b/aHt+1us99tpx/avby9)
Uncaught (in promise) 
Object { code: "rest_no_route", message: "No route was found matching the URL and request method.", data: {…} }

The character encoding of a framed document was not declared. The document may appear different if viewed without the document framing it. [2e6e6608-a57f-42c7-a8fe-1d50f5841998](blob:https://SITE_ADDRESS/2e6e6608-a57f-42c7-a8fe-1d50f5841998)
Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content. [2e6e6608-a57f-42c7-a8fe-1d50f5841998](blob:https://SITE_ADDRESS/2e6e6608-a57f-42c7-a8fe-1d50f5841998)
all-css-2-2 was added to the iframe incorrectly. Please use block.json or enqueue_block_assets to add styles to the iframe. 
<link id="all-css-2-2" crossorigin="anonymous" rel="stylesheet" href="https://s0.wp.com/_stati…0XSc8AhA==&cssminify=yes" type="text/css" media="all">
[index.min.js:10:75810](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/block-editor/index.min.js?m=1686787612i&ver=71ed89ca77d0a313783a)
all-css-4-2 was added to the iframe incorrectly. Please use block.json or enqueue_block_assets to add styles to the iframe. 
<link id="all-css-4-2" crossorigin="anonymous" rel="stylesheet" href="https://s0.wp.com/_stati…fr/QVvMbWe&cssminify=yes" type="text/css" media="all">
[index.min.js:10:75810](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/block-editor/index.min.js?m=1686787612i&ver=71ed89ca77d0a313783a)
all-css-6-2 was added to the iframe incorrectly. Please use block.json or enqueue_block_assets to add styles to the iframe. 
<link id="all-css-6-2" crossorigin="anonymous" rel="stylesheet" href="https://s0.wp.com/_stati…0EGbG4Vw==&cssminify=yes" type="text/css" media="all">
[index.min.js:10:75810](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/block-editor/index.min.js?m=1686787612i&ver=71ed89ca77d0a313783a)
This page uses the non standard property “zoom”. Consider using calc() in the relevant property values, or using “transform” along with “transform-origin: 0 0”. [2e6e6608-a57f-42c7-a8fe-1d50f5841998](blob:https://SITE_ADDRESS/2e6e6608-a57f-42c7-a8fe-1d50f5841998)
Source map error: Error: request failed with status 404
Resource URL: https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/block-editor/index.min.js?m=1686787612i&ver=71ed89ca77d0a313783a
Source Map URL: index.min.js.map

Partitioned cookie or storage access was provided to “https://widgets.wp.com/notifications/?v=calypso-4917-13218-g32921294e7&locale=en#https://SITE_ADDRESS/wp-admin/site-editor.php” because it is loaded in the third-party context and dynamic state partitioning is enabled.

InstallTrigger is deprecated and will be removed in the future. [rest-proxy:16:5020](https://public-api.wordpress.com/wp-admin/rest-proxy/?v=2.0#https://widgets.wp.com)
MouseEvent.mozPressure is deprecated. Use PointerEvent.pressure instead. [index.min.js:10:75299](https://s0.wp.com/wp-content/plugins/gutenberg-core/v16.0.0/build/block-editor/index.min.js?m=1686787612i&ver=71ed89ca77d0a313783a)
```

</details>

**Comment by mattwiebe** [comment]
Thanks @cuemarie - I have been able to duplicate this now. The requirements are:

1. A block theme that enables the Site Editor
2. Default homepage
3. Private site (_not_ Coming Soon, but Private)
4. A mapped domain

Fixing this will not be that simple (ha) and it's not very high on my priority list. In the meantime, mitigation steps could include:

1. Manually load the desired Home template (Templates --> Home)
2. Temporarily set the `*.wordpress.com` domain to primary
3. Temporarily set the site to public

**Comment by cuemarie** [comment]
Thank you, @mattwiebe ! I think given the ease of workaround 1, this shouldn't be too impactful on users. It might make for some unnecessary contacts to support while it's open, but we can keep track of those, and it'll be great to have a clear sense of the expectations HEs can set for future reports in the meantime!

**Comment by cuemarie** [comment]
📌 ACTIONS
- Moved to `Non-Urgent` 

📌 MESSAGE TO HEs
- This fix will likely wait until other higher priority issues are addressed. Please assist users with finding a workaround that fits their needs, and set expectations accordingly!

**Comment by mattwiebe** [comment]
@cuemarie this should be resolved with D115999-code - please do verify and re-open if needed but I think this should put it to bed

**Comment by cuemarie** [comment]
Thank you @mattwiebe ! Just ran a test on a Private Simple Site + custom mapped domain as primary + default template as homepage, and editor loaded properly! 🎉
